### DISCUSSION TITLE
Welcome to Discussions!

### DISCUSSION THREAD

**Discussion by jpeterson** [EsriPS/arcgis-assistant-feedback#1] [General]
## 👋 Welcome!
  We’re using GitHub Discussions as a place to connect with users of ArcGIS Assistant. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded.

To get started, feel free to comment below and introduce yourself! Or, if you're ready, [start a new discussion topic](https://github.com/EsriPS/arcgis-assistant-feedback/discussions/new).

**Comment by DMFACER** [comment]
We rely on the tools currently available at https://ago-assistant.esri.com/ and utilize them on a regular basis.  It looks like some of the same functionality is starting to appear in this new ArcGIS Assistant tool.  Is the plan to make all the tools available in the old assistant available in this new one?  The most critical tool for us is the update URLs of services in a web map.  

At first glance, I do like the new interface in ArcGIS Assistant.

**Reply by slibby** [reply]
Hi @DMFACER The goal is to retain the majority of functionality - what you raise is one example with a bit more nuance. The functionality to replace URLs in a web map is available now in Assistant through editing Item JSON, what's not available is the more "automated" matching of URLs to new URLs. We'll be evaluating interest in these kind of functionality through the Beta. Thanks for your interest!

**Reply by plarkin** [reply]
Updating URLs is the primary reason I use AGO Assistant

**Reply by berniejconnors** [reply]
Same here - Updating URLs is the primary reason I use AGO Assistant.

The "Common Workflows" in the ArcGIS Assistant User Guide should be updated to show the process for Updating URLs in web maps.

As it stands now the AGO Assistant is the better tool to update web map URLs.

**Reply by anninahw** [reply]
Definitely a needed feature, as the classic ago-assistant doesn't recognise grouped layers... you have to move them out of the group to make them appear in the layer list.
So probably search and replace in the json and risk to mess up the code...

**Comment by rlesperance** [comment]
Josh and co.  Thanks for taking this on!  I'm wondering if you guys are working on the full copy of feature layers?  Right now it appears reference only, is that correct?  Can we expect that functionality soon?
Thanks

**Reply by jpeterson** [reply]
Hey @rlesperance, you're correct that currently Assistant only copies layers by reference. 

Deep copying (aka cloning) layers is definitely something we plan to cover in the near future! In the meantime, I would recommend continuing to use https://ago-assistant.esri.com  for this capability.

**Reply by raynaldgarnier** [reply]
Hi @jpeterson! I'm reaching you to find out if there are still plans to implement **Full item copy** in ArcGIS Assistant. The original service replication would really be useful in many cases. Thanks a lot in advance for your reply. 🙂

**Reply by jpeterson** [reply]
Hi @raynaldgarnier :) Yes! @thollingshead and @JPStupfel are working on getting this into Assistant.

**Reply by raynaldgarnier** [reply]
That's excellent news! Can't wait to test this. Thanks a lot!

**Reply by AndreasSour76** [reply]
Hi @jpeterson !

I am trying to copy items (feature layers) from AGOL to enterprise. When I am using the copy function I can clone the feature layer to enterprise, the geometry of the item follows and seems right as well as the attribute structure, however the attributes information is missing. I always get empty attributes. Does the copy function support the cloning of attribute information? 

Thank you in advance!

/Andreas

**Reply by jpeterson** [reply]
Hi @AndreasSour76, copying features should include the cloning of attribute values, so this sounds like an issue we haven't run into before. Could you share more info that might help us reproduce the issue? The version of ArcGIS Enterprise would be helpful, as well as anything particular about the feature layer(s) you're copying.

Thanks!

cc @thollingshead

**Reply by AndreasSour76** [reply]
Hi @jpeterson Thank you for your quick reply! 

I am testing to clone mostly feature layers from ArcGIS Online to our Enterprise (11.2). I am testing with different feature layers, produced by different departments in our organisation, some features include just one layer and some other multiple layers as well as some with simple and others with more complex attribute structure. What I observed is that hosted feature layers have problem to clone attribute values, they always comes up empty, even though the geometry is correct. The non hosted features layers can be cloned without any problem however they are not able to edit. 

The reason I am making those tries is because our organisation will migrate all items from our current AGOl and old Enterprise to a new Enterprise (11.3) the upcoming months and I am trying to find out a suitable and relatively easy way for our users to migrate their data to the new Enterprise.

**Reply by slibby** [reply]
Can you try a really simple feature layer with some simple attributes first? And see if that is also getting dropped? Where we have seen attribute copying issues, it most often relates to column name issues or the contents of that attribute including unexpected characters or HTML tags.

**Comment by JoeDBryant** [comment]
Hi, I work for a utility consulting firm. We are finding ourselves standing up a web GIS for clients during projects, and then needing a solution to hand off the ownership of those services to the client at the end of the project. We also have an internal enterprise portal and external ArcGIS Online organization that we need to copy items between when they are in development. I am hoping this tool will help get these jobs done more efficiently.

**Reply by jpeterson** [reply]
Hey @JoeDBryant, hopefully you find Assistant helpful!
